### PR TITLE
Documentation/interface convention

### LINKED ISSUE #337: Clarification of other places for EnvironmentalConditions
Currently, there are `UNKNOWN` levels of different environmental conditions, e.g.

```
        // Intensity of precipitation is unknown (must not be used in ground
        // truth).
        //
PRECIPITATION_UNKNOWN = 0;
``` 

in 
https://github.com/OpenSimulationInterface/open-simulation-interface/blob/f9b13826c360b4a492b3a84d4dd55b2d5566d85d/osi_environment.proto#L82 ,

which is not to be used in ground truth. Is there another place in which `message EnvironmentalConditions` appears outside of the ground truth message?

### PR DESCRIPTION
**Pull request by vkresch**
#### Reference to a related issue in the repository
resolves #337 

#### Add a description
Adding documentation to message, message fields, enum and enum fields.

#### Mention a member
@jdsika pls review:)

#### Check the checklist

- [x] My code and comments follow the [style guidelines](https://opensimulationinterface.github.io/osi-documentation/osi/commenting.html) and [contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/osi/howtocontribute.html) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the [documentation](https://github.com/OpenSimulationInterface/osi-documentation).
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests / travis ci pass locally with my changes.

### CODE REVIEW

File: doc/interfaceconventions.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+.. code-block:: protobuf
+
+    message EnvironmentalConditions
+    {
+    }
+
+Field Message Naming
+---------------------
+After defining a message custom fields can be added to it in snake case format. This means every letter is lower case and the words are connected through an underline character. See example below:
```
Comment (pmai) [inline doc/interfaceconventions.rst]:
```suggestion
After defining a message fields can be added to it in snake case format. This means every letter is lower case and the words are connected through an underline character. See example below:
```

File: doc/interfaceconventions.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Message Naming
+---------------
+A message definition should always be in camel case. This means that the first letter of each word in a message should be upper case without any spaces. See example below:
+
+.. code-block:: protobuf
+
+    message EnvironmentalConditions
+    {
+    }
+
```
Comment (pmai) [inline doc/interfaceconventions.rst]:
```suggestion

Top-Level Messages
-------------------
All messages that are intended to be exchanged as a stand-alone message, i.e. not required to be embedded in another message, like e.g. ``SensorView`` or ``SensorViewConfiguration``, are required to carry an ``InterfaceVersion`` field as their first field, and a ``Timestamp`` field as their second field, e.g.:

.. code-block:: protobuf

    message TopLevel
    {
        // The interface version used by the sender (simulation environment).
        //
        optional InterfaceVersion version = 1;
        
        // The data timestamp of the simulation environment. Zero time is arbitrary
        // but must be identical for all messages. Zero time does not need to
        // coincide with the UNIX epoch. Recommended is the starting time point of
        // the simulation.
        //
        optional Timestamp timestamp = 2;
    }

```

File: doc/interfaceconventions.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+---------------------
+After defining a message custom fields can be added to it in snake case format. This means every letter is lower case and the words are connected through an underline character. See example below:
+
+.. code-block:: protobuf
+
+    message EnvironmentalConditions
+    {
+        optional double atmospheric_pressure = 1;
+    }
+
```
Comment (pmai) [inline doc/interfaceconventions.rst]:
```suggestion

Field Numbering
----------------

Fields should be numbered consecutively  starting from 1 on first definition. During maintenance, the rules of backward/forward-compatibility require that fields are never renumbered, and field numbers never re-used unless a major release is performed.

All field numbers of 10000 and above are reserved for user-defined extensions, and will thus never be used by OSI message fields.

```

File: doc/interfaceconventions.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        optional double atmospheric_pressure = 1;
+
+        enum AmbientIllumination
+        {
+        }
+    }
+
+Enum Field Naming
+------------
+The naming of an enum field should be all in upper case. The start should be converted from the enum name camel case to upper case snake case. It is mandatory to add to the first enum field name the postfix ``_UNKNOWN`` and to the second the postfix ``_OTHER``. After that the naming can be decided by the user. It is often mentioned that the value ``_UNKNOWN`` should not be used in a ``GroundTruth`` message as there are now uncertanties by definition in ``the truth``. These values are mostly used in messages like ``SensorData`` where the content is subject to interpretation. See example below:
```
Comment (goretkin) [inline doc/interfaceconventions.rst]: `now uncertanties` ->  `no uncertainties`

**Pull request merged**